File: leetcode2399.java
class Solution {
    public boolean checkDistances(String s, int[] distance) {
      HashMap<Character,Integer>map = new HashMap<Character,Integer>();
      for(int i = 0;i<s.length();i++){
        if(!map.containsKey(s.charAt(i))){
          map.put(s.charAt(i),i);
        }else{
          map.put(s.charAt(i),i - 1 - map.get(s.charAt(i)));
        }
      }
      for(Character ch : map.keySet()){
        if(distance[ch - 'a'] != map.get(ch)){
          return false;
        }
      }
      return true;
    }
}

File: leetcode2399.cpp
class Solution {
public:
    bool checkDistances(string s, vector<int>& distance) {
      int arr[26] = {0};
      for(int i = 0;i<s.length() - 1;i++){
        for(int j = i + 1;j<s.length();j++){
          if(s[i] == s[j]){
            if(distance[s[i] - 'a'] != j - i - 1){
              return false;
            }
          }
        }
      }
      return true;
    }
};
